### PR TITLE
Fix MediaBrowsers.Providers analysis issues

### PR DESCRIPTION
**Pull request by richlander**
Resolves all analysis issues for MediaBrowsers.Providers project.

Supports https://github.com/jellyfin/jellyfin/issues/2149

### CODE REVIEW

**Comment by crobibero** [comment]
Thank you for the PR. If you continue making these fixes, please split the PR up into parts so reviewing is easier

**Comment by richlander** [comment]
How would you like me to do that? Like 20 fixes at a time or number of files?

**Comment by crobibero** [comment]
How I typically organize it is to remove x number of warnings, each warning in a commit

see https://github.com/jellyfin/jellyfin/pull/6281 for example

**Comment by richlander** [comment]
Oh, I see. I never thought about doing it that way.

I see I have 24 more errors to fix for the build to pass. If it's OK, I'll do fix those, and then switch to your approach for the next one.

**Comment by crobibero** [comment]
That works for me, thanks again

**Review by crobibero — APPROVED** [review]

**Review by barronpm — APPROVED** [review]

File: MediaBrowser.Providers/Plugins/Tmdb/TmdbClientManager.cs [hunk 1]
```diff
@@ -532,5 +532,23 @@ private Task EnsureClientConfigAsync()
         {
             return !_tmDbClient.HasConfig ? _tmDbClient.GetConfigAsync() : Task.CompletedTask;
         }
+
+        /// <summary>Dispose method.</summary>
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/Tmdb/TmdbClientManager.cs:536]: Same

File: MediaBrowser.Providers/Plugins/Tmdb/TmdbClientManager.cs [hunk 2]
```diff
@@ -532,5 +532,23 @@ private Task EnsureClientConfigAsync()
...
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>IDispose implementation.</summary>
+        /// <param name="disposing">Specify true to dispose.</param>
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/Tmdb/TmdbClientManager.cs:544]:
```suggestion
/// <summary>
        /// Releases unmanaged and - optionally - managed resources.
        /// </summary>
        /// <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
```

File: MediaBrowser.Providers/Subtitles/SubtitleManager.cs [hunk 3]
```diff
@@ -252,8 +252,14 @@ private async Task TrySaveToFiles(Stream stream, List<string> savePaths)
                 }
                 catch (Exception ex)
                 {
-                    (exs ??= new List<Exception>()).Add(ex);
-                }
+#pragma warning disable CA1508
```
Comment (Bond-009) [inline MediaBrowser.Providers/Subtitles/SubtitleManager.cs:255]: Maybe add a note that this is a bug in the analyzer?
Could also add a ref
https://github.com/dotnet/roslyn-analyzers/issues/5160

**Review by Bond-009 — CHANGES_REQUESTED** [review]

**Comment by richlander** [comment]
I believe I resolved the outstanding feedback. Thanks!

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**